src/IR.cpp

```diff
@@ -245,15 +246,23 @@ void IR::saveJSON(std::ostream& Out) const {
   MessageType Message;
   this->toProtobuf(&Message);
   std::string S;
-  google::protobuf::util::MessageToJsonString(Message, &S);
-  Out << S;
+  auto status = google::protobuf::util::MessageToJsonString(Message, &S);
+  if (!status.ok()) {
+    Out << status;
```

Comment:
Hey @5c4lar, apologies that this has taken so long, but we are going through a review internally to incorporate these changes.

Did you have some case where you are able to cause `MessageToJsonString` to fail in this function? Or is there something else specifically that motivated this change?

I welcome the defensive error handling, but just want to understand the potential failure modes here.

Reply:
Just to prevent compile time warnings, I haven't encountered any failure.